bugsnag-android-core/src/main/java/com/bugsnag/android/ConnectivityCompat.kt

```diff
@@ -31,10 +31,23 @@ internal class ConnectivityCompat(
             else -> ConnectivityLegacy(context, cm, callback)
         }
 
-    override fun registerForNetworkChanges() = connectivity.registerForNetworkChanges()
-    override fun hasNetworkConnection() = connectivity.hasNetworkConnection()
-    override fun unregisterForNetworkChanges() = connectivity.unregisterForNetworkChanges()
-    override fun retrieveNetworkAccessState() = connectivity.retrieveNetworkAccessState()
+    override fun registerForNetworkChanges() {
+        runCatching { connectivity.registerForNetworkChanges() }
+    }
+
+    override fun hasNetworkConnection(): Boolean {
+        val result = runCatching { connectivity.hasNetworkConnection() }
+        return result.getOrElse { true } // allow network requests to be made if state unknown
```

Comment:
This felt like a reasonable trade-off as we would only need to catch any `RuntimeException` in one place rather than two, and it minimises the scope of the change. If you feel strongly about this I can add individual checks to `ConnectivityLegacy` and `ConnectivityApi24` instead.

Reply:
No strong objection, just an opinion so feel free to disregard.